streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -37,7 +37,14 @@ import zio.duration.Duration
  *
  */
 trait ZStream[-R, +E, +A] extends Serializable { self =>
-  import ZStream.Fold
+  import ZStream.{ Fold, InputStream }
+
+  /**
+   * Obtain a managed input stream that can be used to read from the stream until it is
+   * empty (or possibly forever, if the stream is infinite). The provided `InputStream`
+   * is valid only inside the scope of the managed resource.
+   */
+  def process: ZManaged[R, E, InputStream[E, A]] = processDefault
```

Comment:
The way I understood @iravid was that we keep this abstract and put process = processDefault on all anonymous subclasses that we create here.

Reply:
Yes, it'll have to be done sooner or later.